service-loadbalancer/service_loadbalancer.go

```diff
@@ -326,6 +335,9 @@ func (lbc *loadBalancerController) getEndpoints(
 	// the target port are capable of service traffic for it.
 	for _, ss := range ep.Subsets {
 		for _, epPort := range ss.Ports {
+			if epPort.Protocol == api.ProtocolUDP {
```

Comment:
yes because without this a service using TCP and UDP result in two elements in the array with the same ip:port.